File: Java/DesignPatterns/src/test/java/designPatterns/structural/flyweight/MainFlyweightTest.java
package designPatterns.structural.flyweight;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainFlyweightTest {
  private ConcreteFlyweight concreteFly;

  @BeforeEach
  void setUp() {
    concreteFly = (ConcreteFlyweight) FlyweightFactory.getConcreteFlyweight("value");
  }

  @Test
  void equalsTest() {
    assertThat(concreteFly.doSomething()).isEqualTo("ConcreteFlyweight: value");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/structural/flyweight/MainFlyweightTest.kt
package structural.flyweight

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainFlyweightTest {
  private lateinit var concreteFly: ConcreteFlyweight

  @BeforeEach
  internal fun setUp() {
    concreteFly = FlyweightFactory.getConcreteFlyweight("value") as ConcreteFlyweight
  }

  @Test
  internal fun equalsTest() {
    assertThat(concreteFly.doSomething()).isEqualTo("ConcreteFlyweight: value")
  }
}